src/scaling_of_discrete_log.rs

```diff
@@ -0,0 +1,856 @@
+// Author: dWallet Labs, Ltd.
+// SPDX-License-Identifier: BSD-3-Clause-Clear
+use std::marker::PhantomData;
+
+use crypto_bigint::{Encoding, Uint};
+use group::{direct_product, GroupElement, KnownOrderGroupElement, PrimeGroupElement};
+use homomorphic_encryption::{AdditivelyHomomorphicEncryptionKey, GroupsPublicParametersAccessors};
+use maurer::{language::GroupsPublicParameters, Error, SOUND_PROOFS_REPETITIONS};
+use serde::{Deserialize, Serialize};
+
+use crate::{language::DecomposableWitness, EnhanceableLanguage};
+
+/// Scaling of a Discrete Log Maurer Language
+///
+/// SECURITY NOTICE:
+/// This language implicitly assumes that the plaintext space of the encryption scheme and the
+/// scalar group coincide (same exponent). Using generic encryption schemes is permitted if and only
+/// if we use this language in its enhanced form, i.e. `EnhancedLanguage`.
+///
+/// SECURITY NOTICE (2):
+/// Furthermore, even when using `EnhancedLanguage`, note that ENC_DH proves a correct computation
+/// that is not a secure function evaluation. That is, the result is not safe to decrypt, as it does
+/// not hide the number of arithmetic reductions mod q. For secure function evaluation, use
+/// `DComEval` (enhanced) language. Because correctness and zero-knowledge is guaranteed for any
+/// group and additively homomorphic encryption scheme in this language, we choose to provide a
+/// fully generic implementation.
+///
+/// However knowledge-soundness proofs are group and encryption scheme dependent, and thus we can
+/// only assure security for groups and encryption schemes for which we know how to prove it.
+///
+/// In the paper, we have proved it for any prime known-order group; so it is safe to use with a
+/// `PrimeOrderGroupElement`.
+///
+/// In regards to additively homomorphic encryption schemes, we proved it for `paillier`.
+#[derive(Clone, Serialize, Deserialize, PartialEq, Debug, Eq)]
+pub struct Language<
```

Comment:
Let's add like we did with after the audit, in the comments a mapping between the variable names in the code and in the paper. Also let's describe the language definition here mathematically as well.